llmfoundry/callbacks/generate_callback.py

```diff
@@ -74,9 +75,11 @@ def generate(self, state: State, logger: Logger):
         dummy_input = device.tensor_to_device(dummy_input)
         with get_precision_context(state.precision):
             with torch.no_grad():
+                assert isinstance(model.model, torch.nn.Module)
                 _ = model.model(input_ids=dummy_input)
 
-            output_token_ids = model.model.generate(
+            assert hasattr(model.model, 'generate')
```

Comment:
This might want to be a `ValueError`, since a user could conceivably try this callback with a model that doesn't have generate function?

Reply:
Yeah, thats fair. Will change.